sema/import_test.go

```diff
@@ -824,3 +824,237 @@ func TestCheckImportContract(t *testing.T) {
+					},
+				},
+			},
+		)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("invalid duplicate aliases", func(t *testing.T) {
+
+		importedChecker, err := ParseAndCheckWithOptions(t,
+			`
+            access(all) fun a(): Int {
+              return 42
+          	}
+
+			access(all) fun b(): Int {
+              return 50
+          	}
+			  
+			`,
+			ParseAndCheckOptions{
+				Location: ImportedLocation,
+			},
+		)
+
+		require.NoError(t, err)
+
+		_, err = ParseAndCheckWithOptions(t,
+			`
+            import a as c from "imported"
+			import b as c from "imported"
+
+            access(all) fun main() {
+                c() + c()
+            }
+            `,
+			ParseAndCheckOptions{
+				Config: &sema.Config{
+					ImportHandler: func(_ *sema.Checker, _ common.Location, _ ast.Range) (sema.Import, error) {
+						return sema.ElaborationImport{
+							Elaboration: importedChecker.Elaboration,
+						}, nil
+					},
+				},
+			},
+		)
+
+		errs := RequireCheckerErrors(t, err, 1)
+
+		redeclarationError := &sema.RedeclarationError{}
+		assert.ErrorAs(t, errs[0], &redeclarationError)
+
+	})
+
+	t.Run("invalid missing aliased import", func(t *testing.T) {
+
+		importedChecker, err := ParseAndCheckWithOptions(t,
+			`
+            access(all) fun a(): Int {
+              return 42
+          	}
+			  
+			`,
+			ParseAndCheckOptions{
+				Location: ImportedLocation,
+			},
+		)
+
+		require.NoError(t, err)
+
+		_, err = ParseAndCheckWithOptions(t,
+			`
+            import c as a from "imported"
+
+            access(all) fun main() {
+                c() + c()
+            }
+            `,
+			ParseAndCheckOptions{
+				Config: &sema.Config{
+					ImportHandler: func(_ *sema.Checker, _ common.Location, _ ast.Range) (sema.Import, error) {
+						return sema.ElaborationImport{
+							Elaboration: importedChecker.Elaboration,
+						}, nil
+					},
+				},
+			},
+		)
+
+		errs := RequireCheckerErrors(t, err, 1)
+
+		notExportedError := &sema.NotExportedError{}
+		assert.ErrorAs(t, errs[0], &notExportedError)
+
+	})
+
+	t.Run("invalid use orig instead of alias", func(t *testing.T) {
+
+		importedChecker, err := ParseAndCheckWithOptions(t,
+			`
+            access(all) fun a(): Int {
+              return 42
+          	}
+			  
+			`,
+			ParseAndCheckOptions{
+				Location: ImportedLocation,
+			},
+		)
+
+		require.NoError(t, err)
+
+		_, err = ParseAndCheckWithOptions(t,
+			`
+            import a as b from "imported"
+
+            access(all) fun main() {
+                a()
+            }
+            `,
+			ParseAndCheckOptions{
+				Config: &sema.Config{
+					ImportHandler: func(_ *sema.Checker, _ common.Location, _ ast.Range) (sema.Import, error) {
+						return sema.ElaborationImport{
+							Elaboration: importedChecker.Elaboration,
+						}, nil
+					},
+				},
+			},
+		)
+
+		errs := RequireCheckerErrors(t, err, 1)
+
+		notDeclaredError := &sema.NotDeclaredError{}
+		assert.ErrorAs(t, errs[0], &notDeclaredError)
+
+	})
+
```

Comment:
Can you please add a test checking that `getType().identifier` still returns the original type name, and does not use the alias?

Reply:
Added here https://github.com/onflow/cadence/pull/3717/files#diff-501230d8bbe927ccf68df43c9ce8dbb3cc6cde1355abb293bd87bbfbbf3fbee6R546.